From cb851cfe883af568e56e50b492579b49869928ab Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <me@zygoon.pl>
Date: Thu, 17 Jan 2019 15:48:46 +0200
Subject: [PATCH 1/9] cmd/snap-seccomp: use upstream seccomp package

Upstream snapd uses a fork that carries additional compatibility patch
required to build snapd for Ubuntu 14.04. This patch is not required with
the latest snapshot of the upstream seccomp golang bindings but they are
neither released upstream nor backported (in their entirety) to Ubuntu
14.04.

The forked seccomp library is not packaged in Debian. As such, to build
snapd, we need to switch to the regular, non-forked package name.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>
Signed-off-by: Maciej Borzecki <maciej.zenon.borzecki@canonical.com>
---
 cmd/snap-seccomp/main.go             | 5 +----
 cmd/snap-seccomp/main_test.go        | 2 +-
 cmd/snap-seccomp/versioninfo.go      | 2 +-
 cmd/snap-seccomp/versioninfo_test.go | 2 +-
 4 files changed, 4 insertions(+), 7 deletions(-)

Index: snapd/cmd/snap-seccomp/main.go
===================================================================
--- snapd.orig/cmd/snap-seccomp/main.go
+++ snapd/cmd/snap-seccomp/main.go
@@ -180,10 +180,7 @@ import (
 	"strings"
 	"syscall"
 
-	// FIXME: we want github.com/seccomp/libseccomp-golang but that
-	// will not work with trusty because libseccomp-golang checks
-	// for the seccomp version and errors if it find one < 2.2.0
-	"github.com/mvo5/libseccomp-golang"
+	"github.com/seccomp/libseccomp-golang"
 
 	"github.com/snapcore/snapd/arch"
 	"github.com/snapcore/snapd/osutil"
Index: snapd/cmd/snap-seccomp/main_test.go
===================================================================
--- snapd.orig/cmd/snap-seccomp/main_test.go
+++ snapd/cmd/snap-seccomp/main_test.go
@@ -32,7 +32,7 @@ import (
 
 	. "gopkg.in/check.v1"
 
-	"github.com/mvo5/libseccomp-golang"
+	"github.com/seccomp/libseccomp-golang"
 
 	"github.com/snapcore/snapd/arch"
 	main "github.com/snapcore/snapd/cmd/snap-seccomp"
Index: snapd/cmd/snap-seccomp/versioninfo.go
===================================================================
--- snapd.orig/cmd/snap-seccomp/versioninfo.go
+++ snapd/cmd/snap-seccomp/versioninfo.go
@@ -25,7 +25,7 @@ import (
 	"os"
 	"strings"
 
-	"github.com/mvo5/libseccomp-golang"
+	"github.com/seccomp/libseccomp-golang"
 
 	"github.com/snapcore/snapd/cmd/snap-seccomp/syscalls"
 	"github.com/snapcore/snapd/osutil"
Index: snapd/cmd/snap-seccomp/versioninfo_test.go
===================================================================
--- snapd.orig/cmd/snap-seccomp/versioninfo_test.go
+++ snapd/cmd/snap-seccomp/versioninfo_test.go
@@ -25,7 +25,7 @@ import (
 
 	. "gopkg.in/check.v1"
 
-	"github.com/mvo5/libseccomp-golang"
+	"github.com/seccomp/libseccomp-golang"
 
 	main "github.com/snapcore/snapd/cmd/snap-seccomp"
 	"github.com/snapcore/snapd/osutil"
